lib/Test/Nginx/Util.pm

```diff
@@ -1238,32 +1238,42 @@ sub parse_headers ($) {
     return \%headers;
 }
 
-sub expand_env_in_text ($$) {
-    my ($text, $name) = @_;
+sub expand_env_in_text ($$$) {
+    my ($text, $name, $rand_ports) = @_;
 
     if (!defined $text) {
         return;
     }
 
-    my $used_ports = { $ServerPort => 1 };
+    my $used_ports = {
+        $ServerPort => 1,
+        map { $_ => 1 } values %$rand_ports
+    };
 
     $text =~ s/\$(TEST_NGINX_[_A-Z0-9]+)/
+        my $repl;
```

Comment:
What does this variable name mean? We need a better name here.

Reply:
The meaning of repl is "replaced string", and now it changed to "expanded_env", here: https://github.com/openresty/test-nginx/pull/109/commits/c884dd99faafdb51bc36c621e8022bed7b4c9e7e